tools/c7n_azure/c7n_azure/resources/sqlmanagedinstance.py

```diff
@@ -0,0 +1,219 @@
+        filtering_properties = key.split('.')
+        for resource in resources:
+            assessments = client.managed_instance_vulnerability_assessments.list_by_instance(
+                resource['resourceGroup'],
+                resource['name']
+            )
+            add_to_filtered = self.filter_by_assessments(assessments, filtering_properties, value)
+            if add_to_filtered:
+                filtered_resources.append(resource)
+
+        return super(SqlManagedInstanceVulnerabilityAssessmentsFilter, self).process(
+            filtered_resources, event)
+
+
+@SqlManagedInstance.filter_registry.register('encryption-protector')
+class SqlManagedInstanceEncryptionProtectorsFilter(Filter):
+    """
+    Filters resources by encryption protectors.
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: azure-sql-managed-instance-service-managed
+            resource: azure.sql-managed-instance
+            filters:
+              - type: encryption-protector
+                key: kind
+                value: servicemanaged
+    """
+
+    schema = type_schema(
+        'encryption-protector', rinherit=ValueFilter.schema
+    )
+
+    def __call__(self, resource):
+        return resource
+
+    @staticmethod
+    def filter_by_encryption_protector(protectors, filtering_properties, value):
+        add_to_filtered = False
+        for protector in protectors:
+            protector = protector.as_dict()
+            property_value = protector
+            for filtering_property in filtering_properties:
+                if filtering_property in property_value:
+                    property_value = property_value[filtering_property]
+                else:
+                    property_value = None
+                    break
+            if isinstance(property_value, bool) and isinstance(value, str):
+                value = bool(strtobool(value))
+            if value == property_value:
+                add_to_filtered = True
+                break
+        return add_to_filtered
+
+    def process(self, resources, event=None):
+        client = self.manager.get_client('azure.mgmt.sql.SqlManagementClient')
+
+        filtered_resources = []
+        key = self.data['key']
+        value = self.data['value']
+        filtering_properties = key.split('.')
+        for resource in resources:
+            protectors = client.managed_instance_encryption_protectors.list_by_instance(
+                resource['resourceGroup'],
+                resource['name']
+            )
+            add_to_filtered = self.filter_by_encryption_protector(
+                protectors, filtering_properties, value)
+
+            if add_to_filtered:
+                filtered_resources.append(resource)
+
+        return super(SqlManagedInstanceEncryptionProtectorsFilter, self).process(
+            filtered_resources, event)
+
+
+@SqlManagedInstance.filter_registry.register('managed-server-security-alert-policies')
```

Comment:
please rename to drop resource in the filer name.

Reply:
renamed to `security-alert-policies`